lending/loan_management/report/future_cashflow_report/future_cashflow_report.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2025, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+
+
+def execute(filters=None):
+	filters = filters or {}
+	columns = get_columns()
+	data = get_data(filters)
+	return columns, data
+
+
+def get_columns():
+	return [
+		{
+			"label": _("Loan"),
+			"fieldname": "loan",
+			"fieldtype": "Link",
+			"options": "Loan",
+			"width": 150,
+		},
+		{
+			"label": _("Loan Disbursement"),
+			"fieldname": "loan_disbursement",
+			"fieldtype": "Link",
+			"options": "Loan Disbursement",
+			"width": 150,
+		},
+		{
+			"label": _("Applicant Name"),
+			"fieldname": "applicant",
+			"fieldtype": "Data",
+			"width": 200,
+		},
+		{
+			"label": _("Loan Product"),
+			"fieldname": "loan_product",
+			"fieldtype": "Link",
+			"options": "Loan Product",
+			"width": 150,
+		},
+		{
+			"label": _("Payment Date"),
+			"fieldname": "payment_date",
+			"fieldtype": "Date",
+			"width": 120,
+		},
+		{
+			"label": _("Principal Amount"),
+			"fieldname": "principal_amount",
+			"fieldtype": "Currency",
+			"width": 150,
+		},
+		{
+			"label": _("Interest Amount"),
+			"fieldname": "interest_amount",
+			"fieldtype": "Currency",
+			"width": 150,
+		},
+		{
+			"label": _("Total Payment"),
+			"fieldname": "total_payment",
+			"fieldtype": "Currency",
+			"width": 150,
+		},
+	]
+
+
+def get_data(filters):
+	if not filters.get("company"):
+		frappe.throw(_("Please select Company."))
+	if not filters.get("as_on_date"):
+		frappe.throw(_("Please select As on Date."))
+
+	Loan = frappe.qb.DocType("Loan")
+	LoanRepaymentSchedule = frappe.qb.DocType("Loan Repayment Schedule")
+	RepaymentSchedule = frappe.qb.DocType("Repayment Schedule")
+	LoanDisbursement = frappe.qb.DocType("Loan Disbursement")
+
+	query = (
+		frappe.qb.from_(Loan)
+		.join(LoanDisbursement)
```

Comment:
Loan and Loan Disbursement links are both part of the "Loan Repayment Schedule" table. 
Why do you need 4 table joins here? Just the parent and child table join should be okay to select all the required fields. You should be able to judge these obvious things by yourself now.